### ISSUE TITLE
1.1.126+: False positive for | operator between matching sets of literals

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#4436]
This issue arises starting in pyright 1.1.126.

```python
from typing import Set
from typing_extensions import Literal

A = Literal["A", "B"]

def f() -> Set[A]: ...

def g(x: Set[A]): ...

# error: Operator "|" not supported for types "Set[A]" and "Set[A]"
#    Operator "|" not supported for types "Set[A]" and "Set[A]" when expected type is "Set[A]" (reportGeneralTypeIssues)
# 1 error, 0 warnings, 0 informations
g(f() | f())
```

**Comment by ottumm** [comment]
I've been seeing this same error, although not for literals, starting in Pyright 1.1.286.

```
error: Operator "|" not supported for types "set[StackupLayer]" and "set[StackupLayer]"
[34](https://github.com/dangerdevices/florist/actions/runs/3897162379/jobs/6654541383#step:4:35)
    Operator "|" not supported for types "set[StackupLayer]" and "set[StackupLayer]" when expected type is "set[StackupLayer]" (reportGeneralTypeIssues)
```

**Comment by ottumm** [comment]
Notably, I'm seeing this in Pyright 1.1.288 but _not_ in the latest version of Pylance in VS Code.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.289, which I just published. It will also be included in a future release of pylance.

**Comment by ottumm** [comment]
I'm now seeing this in the latest stable Pylance in VS Code but _not_ in Pyright 1.1.289. The bug seems to have flipped.

**Comment by erictraut** [comment]
Which version of pylance are you using? A new version is scheduled to be released today, and I think that's the version that is in sync with pyright 1.1.289. Please wait for a few hours and then install the latest version of pylance. If you don't want to wait, you could download last week's "insiders build" of pylance.

**Comment by ottumm** [comment]
Ha, nice, I just upgraded and that fixed it, thanks!
